DragaliaAPI/Features/DmodeDungeon/DmodeDungeonService.cs

```diff
@@ -58,18 +58,20 @@ await dmodeRepository.Charas.SingleOrDefaultAsync(x => x.CharaId == charaId)
             x => x.CharaId == charaId
         );
 
+        CharaData charaData = MasterAsset.CharaData[charaId];
+
         AtgenUnitData unitData =
             new(
                 chara.CharaId,
-                chara.Skill1Level,
-                chara.Skill2Level,
-                chara.Ability1Level,
-                chara.Ability2Level,
-                chara.Ability3Level,
-                chara.ExAbilityLevel,
-                chara.ExAbility2Level,
-                chara.BurstAttackLevel,
-                chara.ComboBuildupCount
+                charaData.HasManaSpiral ? 4 : 3,
+                charaData.HasManaSpiral ? 3 : 2,
+                charaData.MaxAbility1Level,
+                charaData.MaxAbility2Level,
+                charaData.MaxAbility3Level,
+                5,
+                5,
+                2,
+                0
```

Comment:
ComboBuildupCount = upgraded standard attack damage, this should be 1 for spiral units